src/core/Akka.Remote.Tests/Transport/HeliosTransportDnsResolutionSpec.cs

```diff
@@ -158,6 +166,56 @@ public Property HeliosTransport_Should_Resolve_DNS(EndPoint inbound, EndPoint ou
             }
         }
 
+        [Property]
+        public Property HeliosTransport_Should_Resolve_DNS_with_PublicHostname(IPEndPoint inbound, DnsEndPoint publicInbound,
```

Comment:
Model-based test to verify all combinations of actual IP address + public hostname + DNS resolution preference.